server/command.go

```diff
@@ -184,8 +214,179 @@ func (p *Plugin) ExecuteCommand(_ *plugin.Context, args *model.CommandArgs) (*mo
 
 		return p.getCommandResponse(args, "Settings updated."), nil
 
+	case "webhook":
+		message := p.webhookCommand(parameters, info)
+		response := p.getCommandResponse(args, message)
+		return response, nil
+
+	default:
+		return p.getCommandResponse(args, unknownActionMessage), nil
+	}
+}
+
+// webhookCommand processes the /gitlab webhook commands
+func (p *Plugin) webhookCommand(parameters []string, info *gitlab.GitlabUserInfo) string {
+	if len(parameters) < 1 {
+		return unknownActionMessage
+	}
+	subCommand := parameters[0]
+
+	switch subCommand {
+	case "list":
+		if len(parameters) != 2 {
+			return unknownActionMessage
+		}
+
+		namespace := parameters[1]
+		fullPath := strings.Split(namespace, "/")
+
+		var webhookInfo []*gitlab.WebhookInfo
+		var err error
+		if len(fullPath) == 2 {
+			owner := fullPath[0]
+			repo := fullPath[1]
+
+			webhookInfo, err = p.GitlabClient.GetProjectHooks(info, owner, repo)
+			if err != nil {
+				if strings.Contains(err.Error(), projectNotFoundError) {
+					return projectNotFoundMessage + namespace
+				}
+				return err.Error()
+			}
+
+		} else {
+			owner := fullPath[0]
+			webhookInfo, err = p.GitlabClient.GetGroupHooks(info, owner)
+			if err != nil {
+				if strings.Contains(err.Error(), groupNotFoundError) {
+					return groupNotFoundMessage + owner
+				}
+				return err.Error()
+			}
+		}
+		if len(webhookInfo) == 0 {
+			return fmt.Sprintf("No webhooks found in %s", namespace)
+		}
+		var formatedWebhooks string
+		for _, hook := range webhookInfo {
+			formatedWebhooks += hook.Stringify()
+		}
+		return formatedWebhooks
+
+	case "add":
+		namespace := parameters[1]
+		fullPath := strings.Split(namespace, "/")
+
+		siteURL := *p.API.GetConfig().ServiceSettings.SiteURL
+		urlPath := fmt.Sprintf("%v/%s", siteURL, inboundWebhookURL)
```

Comment:
Shouldn't we add the port as well in case it is not `:80`?

Reply:
Good Thinking, I use port 8065 in my test environment. But I have the port number included in my site URL.

Looking at the System Console under site URL I see, 

> The URL that users will use to access Mattermost. Standard ports, such as 80 and 443, can be omitted, but non-standard ports are required. For example: http://example.com:8065. This setting is required.

So I think it might be safe to assume the port will be included in the Site URL. But if that's not a safe assumption, it should be an easy check.

Reply:
You are absolutely right. One case worth checking might be `siteURL` being empty, because it's a default. I'd suggest using 

`"localhost" + *p.API.GetConfig().ServiceSettings.ListenAddress`

Reply:
I think it's a good idea to check siteURL for empty, I'll add that. 

I'm pretty sure other things won't work (like subscribe and activate) if siteURL is not set. But if gets unset webhook creation would set webhooks to point to just the inboundWebhookURL, which would be unexpected.

I'm not sure computing the URL is practical or desired, they could be behind a load balancer or proxy and we wouldn't know their external IP/FQDN. 

How about I just add a check, and if it is empty I fail the webhook creation with a message like

> Unable to create webhook. The Mattermot Site URL is not set. Set it in the admin console or rerun ```/gitlab webhook add group/project URL``` including the desired URL.